Improve code base quality

alessandropellegrini opened the pull request:
This PR attempts to fix minor nits in the code base. More importantly, I am trying to improve the overall documentation coverage.

alessandropellegrini committed:
Replace pre-C11 __thread with _Thread_local

We still had some __thread storage class specifier in the code base.
Change it to the standard _Thread_local.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Add some const

They are not improving any performance though.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Consistent use of __typeof__

Align __typeof to __typeof__

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
More meaningful variable names

Improve readability by using more readable variable names.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Improve documentation

Documenting several functions/datatypes with missing documentation,
and fixing some documentation errors.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Update email address in CoC

There was an old email address in the code of conduct.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Merge branch 'develop' into code-cleanup

alessandropellegrini committed:
Improve documentation

Documenting several functions/datatypes with missing documentation,
and fixing some documentation errors.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Reorganize the mm subsystem

This commit moves around functions and data structures, based on the
following core ideas:

- Buddy system stuff are confined in mm/buddy/ files
- model_allocator.c should implement everything that can be seen by
  the model or pertains to the LP state memory. It interacts with
  the buddy system, so the concept of "multi.c" is not necessary, nor
  pertains to mm/buddy/. Out of mm/ no one should explicitly see
  the buddy system.
- in mm/checkpoint we have everything that relates to checkpointing of
  the LP state. So we now have an `autonomic.c` unit, a `full.c` unit,
  and an `incremental.c` unit, that is currently only keeping old code.
  `mm/checkpoint/checkpoint.h` is the only header exposing all
  checkpointing-related stuff, so independently of what you need
  (autonomic/full/incremental), you can include this header.

I am convinced this refactor helps people go and find stuff based on
the organization of the tree. There could be a minimal penaly, as we
have lost one `static inline` for a function, but I am happy
tolerating a minimal performance penalty in favor of readability.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Enforce code formatting style

Ensure that all files are adhering to the ROOT-Sim code formatting
style.
There are some exceptions that have been discussed in the past, that
I am keeping as is as they are more readable.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Fix erroneous exported function name

In the mm refactor, a renamed function was not renamed in the
corresponding header.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Pin actions to full-length SHA

This is dumb, because _we_ manage the actions, but yes, our code is
waaaay more secure now.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

Piccions commented on src/mm/buddy/buddy.c:
```diff
@@ -10,84 +10,119 @@
 	/* search recursively for the child */
 	uint_fast8_t node_size = B_TOTAL_EXP;
-	uint_fast32_t i = 0;
+	uint_fast32_t index = 0;
```
please don't use `index` as a variable name as there exists a deprecated POSIX function with the same name (https://man7.org/linux/man-pages/man3/index.3.html). I suggest you use `idx` instead, but feel free to use a different name.

alessandropellegrini replied:
I think it's being removed from the library, but anyhow it makes sense!

Piccions commented on src/mm/buddy/checkpoint.c:
```diff
@@ -1,16 +1,28 @@
 
 
+/**
+ * @brief Traverses the buddy tree and performs an action on each unallocated block.
```
This macro will actually visit allocated blocks (it is used for checkpointing and restore)

Piccions commented on src/lp/process.c:
```diff
@@ -159,51 +183,51 @@ static inline void silent_execution(const struct lp_ctx *lp, array_count_t last_
  * @param past_i the index in @a proc_p of the last validly processed message
  */
-static inline void send_anti_messages(struct process_ctx *proc_p, array_count_t past_i)
+static inline void send_anti_messages(struct process_ctx *lp, const array_count_t past_i)
```
Are you sure you want to name the first argument `lp`? We use `lp` throughout the codebase for variables of type `struct lp_ctx *`, while this is a pointer to an inner component of type `struct process_ctx`. I agree that `proc_p` is ugly, maybe use `pc`?

alessandropellegrini replied:
Silly me, I confused the argument by mistake.

Piccions commented on src/mm/model_allocator.c:
```diff
@@ -0,0 +1,216 @@
+ * SPDX-FileCopyrightText: 2008-2025 HPCS Group <rootsim@googlegroups.com>
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+#include <errno.h>
```
Let's have a consistent order of header inclusions:
1. The header related to the source file followed by a new line (in this case `model_allocator.h`
2. Headers that belong to this project followed by a new line
3. System headers

Piccions reviewed (changes requested):
Good changes, left a couple of comments.
Only thing I am doubtful of is `const` qualification of local variables, but it's not a deal breaker for me

alessandropellegrini committed:
Avoid clashing with index(3)

In stdlib there is the deprecated index(3) function which has been
deprecated in favour of strchr(3). Avoid using index as the name
of the variable to avoid any possible issues.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Fix documentation

Fix wrong documentation.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Fix incorrect parameter name

I confused by mistake the message processing data with an LP context.
This is fixed, and the arguments names are made consistent in the
code base.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini committed:
Make consistent the include order

This is made to align to the project conventions.

Signed-off-by: Alessandro Pellegrini <a.pellegrini@ing.uniroma2.it>

alessandropellegrini commented:
I have fixed all your suggestions, thanks!

Regarding the `const`s, I agree it's not a game changer. I introduced that just to clarify to the reader that the variables are just temporaries that won't be updated.
